src/chainstate/stacks/index/marf.rs

```diff
@@ -481,19 +489,39 @@ impl MARF {
 
         let path = TriePath::from_key(key);
 
-        let leaf_opt_res = MARF::get_path(&mut self.storage, block_hash, &path)?;
+        let result = MARF::get_path(&mut self.storage, block_hash, &path);
 
         // restore
+        // Q: should this return on error? or return a corruption error?
         self.storage.open_block(&cur_block_hash, cur_block_rw)?;
 
-        match leaf_opt_res {
-            None => {
-                Ok(None)
-            },
-            Some(leaf) => {
-                Ok(Some(leaf.data))
-            }
-        }
+        result.map(|option_result| option_result.map(|leaf| leaf.data))
+    }
+
+    pub fn get_block_height(&mut self, block_hash: &BlockHeaderHash, current_block_hash: &BlockHeaderHash) -> Result<Option<u64>, Error> {
+        let hash_key = format!("{}::{}", BLOCK_HASH_TO_HEIGHT_MAPPING_KEY, block_hash);
+
+        self.get(current_block_hash, &hash_key)
+            .map(|option_result| {
+                option_result.map( |marf_value| { u64::from(marf_value) } )
+            })
+    }
+
+    pub fn get_block_at_height(&mut self, height: u64, current_block_hash: &BlockHeaderHash) -> Result<Option<BlockHeaderHash>, Error> {
+        let height_key = format!("{}::{}", BLOCK_HEIGHT_TO_HASH_MAPPING_KEY, height);
+
+        self.get(current_block_hash, &height_key)
+            .map(|option_result| {
+                option_result.map( |marf_value| { BlockHeaderHash::from(marf_value) } )
+            })
+    }
+
+    pub fn set_block_height(&mut self, block_hash: &BlockHeaderHash, height: u64) -> Result<(), Error> {
```

Comment:
Thanks for adding these methods!  Are there any tests for them, as well as `check_block_hash()`?

Reply:
No unit tests -- they get some test coverage from the `at-block` tests, but I can add some unit tests for this.